### DISCUSSION TITLE
How to convert inpaint_v26.fooocus.patch to diffusers lora format?

### DISCUSSION THREAD

**Discussion by peki12345** [lllyasviel/Fooocus#3314] [Q&A] [answered]
I want to merge this lora to diffusers models through the "pipe.fuse_lora()" that comes with the diffusers library, what should I do?

**Comment by IPv6** [comment, marked as answer]
why do you think it should work? inpaint model is not a lora, they works differently afaik

**Reply by mashb1t** [reply]
@peki12345 the inpaint model is no LoRA but of type InpaintHead, see 
https://github.com/lllyasviel/Fooocus/blob/f597bf1ab653bd4bea8cb1e5cc7996348fbe8879/modules/inpaint_worker.py#L13

It's called here
https://github.com/lllyasviel/Fooocus/blob/f597bf1ab653bd4bea8cb1e5cc7996348fbe8879/modules/inpaint_worker.py#L194

I've wrongly called the inpaint model "LoRA" in a previous discussion to emphasize that it's not a checkpoint and a separate thing.
It's not a LoRA but a model loaded by pytorch.
@IPv6 thanks for pointing this out correctly, much appreciated.

**Reply by peki12345** [reply]
> @peki12345 the inpaint model is no LoRA but of type InpaintHead, see
> 
> https://github.com/lllyasviel/Fooocus/blob/f597bf1ab653bd4bea8cb1e5cc7996348fbe8879/modules/inpaint_worker.py#L13
> 
> It's called here
> 
> https://github.com/lllyasviel/Fooocus/blob/f597bf1ab653bd4bea8cb1e5cc7996348fbe8879/modules/inpaint_worker.py#L194
> 
> I've wrongly called the inpaint model "LoRA" in a previous discussion to emphasize that it's not a checkpoint and a separate thing. It's not a LoRA but a model loaded by pytorch. @IPv6 thanks for pointing this out correctly, much appreciated.

Actually, inpaint_v26.fooocus.patch is a lora, but not in the diffusers format. What you mentioned is fooocus_inpaint_head.pth, not inpaint_v26.fooocus.patch. In Fooocus, the process of merging lora is here: https://github.com/lllyasviel/Fooocus/blob/3a20e14ca0bf0e2d4aabd6efa7b9ef9c877e5798/modules/async_worker.py#L892

At the same time, I can successfully merge this lora by saving the model in the diffusers format to a format that Fooocus can accept and executing this code, but this is too troublesome.
